Bug in config.onlyFindTextInsidePaperContour

### There is a bug in `config.onlyFindTextInsidePaperContour`
when tested agianst test file `readableColor.png` we'll get 
`Uncaught AssertionError: expected [ReferenceError: contours is not defined] to equal null`.

 when tested with `config.onlyFindTextInsidePaperContour` and `config.removeNoise` set to true
` Uncaught AssertionError: expected [ReferenceError: contours is not defined] to equal null`

-

#### Further investigation 
Shows there might be a bug upstream in opencv3 when tested  with `config.onlyFindTextInsidePaperContour` and `config.verbose` set to true

```
OpenCV Error: Assertion failed (count >= 0 && (depth == CV_32F || depth == CV_32S)) in arcLength, file /tmp/opencv3-20160929-23220-ldcoax/modules/imgproc/src/shapedescr.cpp, line 285
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: /tmp/opencv3-20160929-23220-ldcoax/modules/imgproc/src/shapedescr.cpp:285: error: (-215) count >= 0 && (depth == CV_32F || depth == CV_32S) in function arcLength

Abort trap: 6
```

`config.onlyFindTextInsidePaperContour`, `config.verbose` and `config.removeNoise` set to true

```
OpenCV Error: Assertion failed (count >= 0 && (depth == CV_32F || depth == CV_32S)) in arcLength, file /tmp/opencv3-20160929-23220-ldcoax/modules/imgproc/src/shapedescr.cpp, line 285
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: /tmp/opencv3-20160929-23220-ldcoax/modules/imgproc/src/shapedescr.cpp:285: error: (-215) count >= 0 && (depth == CV_32F || depth == CV_32S) in function arcLength

Abort trap: 6
```

tests work as expected with `readable.jpg` and `readableGray.png`

-

#### Possible break point `isolatePaper()` 

Changing `cv.Constants.CV_8UC1...cv.Constants.CV_8UC3`
* `cv.Constants.CV_16UC1...cv.Constants.CV_16UC3`
* `cv.Constants.CV_CV_32SC1...cv.Constants.CV_32SC3`
* `cv.Constants.CV_CV_32FC1...cv.Constants.CV_32FC3`

-
##### `readableColor.png`
```
OpenCV Error: Assertion failed (mask.depth() == CV_8U && (mcn == 1 || mcn == cn)) in copyTo, file /tmp/opencv3-20160929-23220-ldcoax/modules/core/src/copy.cpp, line 342
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: /tmp/opencv3-20160929-23220-ldcoax/modules/core/src/copy.cpp:342: error: (-215) mask.depth() == CV_8U && (mcn == 1 || mcn == cn) in function copyTo

Abort trap: 6
```

##### `readableGray.png`
```
OpenCV Error: Assertion failed (mask.depth() == CV_8U && (mcn == 1 || mcn == cn)) in copyTo, file /tmp/opencv3-20160929-23220-ldcoax/modules/core/src/copy.cpp, line 342
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: /tmp/opencv3-20160929-23220-ldcoax/modules/core/src/copy.cpp:342: error: (-215) mask.depth() == CV_8U && (mcn == 1 || mcn == cn) in function copyTo

Abort trap: 6
```

`readable.jpg` work as expected

--- a/lib/image_processor/preprocessor/opencv-photo2scan.js
+++ b/lib/image_processor/preprocessor/opencv-photo2scan.js
@@ -156,11 +156,12 @@ function getTextElementsContours(im) {
     return ratio >= 5;
   });
 
-  return contoursList;
+  return { contours: contours, list: contoursList };
 }
 
 function rotate(paperMask, im, outfile, config) {
-  contoursList = getTextElementsContours(im);
+  contours = getTextElementsContours(im);
+  contoursList = contours.list;
 
   if(config.onlyFindTextInsidePaperContour) {
     // This is not necessary unless the paper haven't been deteced
@@ -168,7 +169,7 @@ function rotate(paperMask, im, outfile, config) {
       var contoursListCountPrevious = contoursList.length;
       // Remove all not inside the paper polygon
       contoursList = contoursList.filter(function(c) {
-        points = getContourPoints(contours, c[0]);
+        points = getContourPoints(contours.contours, c[0]);
         return polygonInPolygon(points, paperMask);
       });
       var removedContoursCount = contoursListCountPrevious - contoursList.length;

